### ISSUE TITLE
GraalVM native-image support for Java, Scala, Kotlin, etc.

### ISSUE THREAD

**Issue by plokhotnyuk** [mattgodbolt/compiler-explorer#1411]

**Comment by mattgodbolt** [comment]
Can  you elaborate? Is this a request for execution support of some new languages? We don't (yet!) support any of those languages. Java support (to intermediate representation) is coming soon in #1396, but we currently have no support for Scala or Kotlin. I'm a little reluctant to add JITted languages for a variety of reasons.

What's "native-image" in this context?

Thanks!

**Comment by plokhotnyuk** [comment]
Yes, it is about adding those languages, but using [AOT compilation](https://www.graalvm.org/docs/reference-manual/aot-compilation/) (not JIT).

**Comment by mattgodbolt** [comment]
Thanks. I'll need to check the GraalVM license allows this.

**Comment by leo60228** [comment]
GraalVM is GPLv2. However, Oracle distributes binaries ("GraalVM EE") with quite a few patches under a very restrictive license. That license seems to unambiguously disallow compiler-explorer's usage:

> Further, You may not: 
> * make the Programs, or any application You develop using the Programs, available in any manner to any third party (other than Contractors acting on Your behalf as set forth in this Agreement); 

I think it's probably fair to include other builds of GraalVM, though they should probably be labeled GraalVM CE for clarity.
